File: backend/src/Biplott.Core/Entities/Game.cs
namespace Biplott.Core.Entities;

public class Game
{
    public int Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? Tagline { get; set; }
    public string? IconUrl { get; set; }
    public bool IsActive { get; set; } = true;
    public int SortOrder { get; set; } = 0;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public List<GamePool> Pools { get; set; } = new();
    public List<Slip> Slips { get; set; } = new();
}

File: frontend/src/types/game.ts
export interface GamePool {
  id: number;
  poolIndex: number;
  name: string;
  minNumber: number;
  maxNumber: number;
  pickCount: number;
  allowDuplicates: boolean;
  badgeColor?: string;
}

export interface Game {
  id: number;
  code: string;
  name: string;
  description: string;
  tagline?: string;
  iconUrl?: string;
  isActive: boolean;
  sortOrder: number;
  pools: GamePool[];
}

export interface ApiResponse<T> {
  success: boolean;
  data: T;
  message: string;
  timestamp: string;
}
